integration-tests/src/test/java/uk/gov/di/authentication/api/LogoutIntegrationTest.java

```diff
@@ -4,37 +4,92 @@
+        String sessionId = "session-id";
+        String clientSessionId = "client-session-id";
         RSAKey signingKey =
                 new RSAKeyGenerator(2048).keyID(UUID.randomUUID().toString()).generate();
         SignedJWT signedJWT =
                 IDTokenGenerator.generateIDToken(
                         "client-id", new Subject(), "http://localhost/issuer", signingKey);
+        RedisHelper.createSession(sessionId);
+        RedisHelper.addAuthRequestToSession(
+                clientSessionId, sessionId, generateAuthRequest().toParameters());
+        RedisHelper.addIDTokenToSession(sessionId, clientSessionId, signedJWT.serialize());
+        DynamoHelper.registerClient(
+                "client-id",
+                "client-name",
+                singletonList("http://localhost:8080/redirect"),
+                singletonList("client-1"),
+                singletonList("openid"),
+                "public-key",
+                singletonList(
+                        "https://di-auth-stub-relying-party-build.london.cloudapps.digital/"));
         Client client = ClientBuilder.newClient();
+        MultivaluedMap<String, Object> headers = new MultivaluedHashMap<>();
+        headers.add(COOKIE, buildCookieString(sessionId, clientSessionId));
         Response response =
                 client.target(ROOT_RESOURCE_URL + LOGOUT_ENDPOINT)
                         .queryParam("id_token_hint", signedJWT.serialize())
-                        .queryParam("post_logout_redirect_uri", "http://localhost/redirect")
+                        .queryParam(
+                                "post_logout_redirect_uri",
+                                "https://di-auth-stub-relying-party-build.london.cloudapps.digital/")
                         .queryParam("state", "8VAVNSxHO1HwiNDhwchQKdd7eOUK3ltKfQzwPDxu9LU")
                         .request()
+                        .headers(headers)
                         .get();
 
         assertEquals(302, response.getStatus());
+        assertTrue(
+                response.getHeaders()
+                        .get("Location")
+                        .contains(
+                                "https://di-auth-stub-relying-party-build.london.cloudapps.digital/?state="
+                                        + "8VAVNSxHO1HwiNDhwchQKdd7eOUK3ltKfQzwPDxu9LU"));
+    }
+
+    private AuthorizationRequest generateAuthRequest() {
+        ResponseType responseType = new ResponseType(ResponseType.Value.CODE);
+        State state = new State();
+        return new AuthorizationRequest.Builder(responseType, new ClientID("test-client"))
+                .redirectionURI(URI.create("http://localhost:8080/redirect"))
+                .state(state)
+                .build();
+    }
+
+    private String buildCookieString(String sessionID, String clientSessionID) {
+        return format(
+                "%s=%s.%s; Max-Age=%d; %s",
```

Comment:
This could perhaps be replaced with something from the standard library like:

```
var cookie = new HttpCookie("gs", "whatever-the-value-is");
cookie.setSecure(true);
cookie.setMaxAge(1800);
cookie.setHttpOnly(true);

return cookie.toString();
```

Reply:
I was thinking similar last week when I saw how the cookie was being generated in the handler.

Reply:
This has been updated as well as the CookieHelper class. I haven't touched how the cookie is being generated for now.